src/types/internal.types.ts

```diff
@@ -24,6 +25,9 @@ export interface ExecutionParams {
   step: Step
   statusManager: StatusManager
   settings: InternalExecutionSettings
+  updateTransactionRequest?: (
+    transactionRequest: TransactionRequest
+  ) => Promise<TransactionRequest>
```

Comment:
Does it make sense to have it as part of the `InternalExecutionSettings` like we have other hooks?

Reply:
Yes this already seems to be used at multiple places. I'm using this in my change.